common/courses/notifications.ts

```diff
@@ -112,7 +112,13 @@ export async function sendPupilCoursePromotion(subcourse: Prisma.subcourse, coun
     }
 
     const pupils = await prisma.pupil.findMany({
-        where: { active: true, isParticipant: true, grade: { in: grades }, subcourse_participants_pupil: { none: { subcourseId: subcourse.id } } },
+        where: {
+            active: true,
+            verification: null,
```

Comment:
What does this line do?

Reply:
Historically, when users registered, we would write a random token into "verification", and then when they verified their email with that token we would NULL it. Later "verifiedAt" was introduced which is the other way round, it is NULL upon registration and then later set. Funnily users could also ignore it, so they could just log in (using a token that was sent to their email) but not verify their email. I somewhen cleaned this up, and now whenever a user logs in (with a token sent to their email) and is not verified yet, we will NULL verified and set verifiedAt. So `verification: null` and `verifiedAt: { not: null }` should be more or less equivalent, though `verification: null` might still be more accurate for historic accounts - we also use it in many other places.

Reply:
So as a quick fix I would bring it in this way, let me create an issue for cleaning up this legacy (we should drop verification, set verifiedAt for all the remaining users, and then consistently use `verifiedAt: { not: null }`

Reply:
Thank you for the explanation 👍